pkg/commands/kustomizationfile_test.go

```diff
@@ -88,3 +88,48 @@ func TestNewNotExist(t *testing.T) {
 		t.Fatalf("expect an error contains %q, but got %v", contained, err)
 	}
 }
+
+func TestPreserveComments(t *testing.T) {
+	kustomizationContentWithComments := []byte(
+		`# shem qing some comments
+# This is some comment we should preserve
+# don't delete it
+resources:
+- pod.yaml
+- service.yaml
+# something you may want to keep
+vars:
+- fieldref:
+    fieldPath: metadata.name
+  name: MY_SERVICE_NAME
+  objref:
+    apiVersion: v1
+    kind: Service
+    name: my-service
+bases:
+- ../namespaces
+# some descriptions for the patches
+patches:
+- service.yaml
+- pod.yaml
```

Comment:
will it preserve the comments on the items in the list ?

Reply:
No, it won't. Only the comments ahead the item is preserved.
```
# some descriptions for the patches
patches:
  # this is service.yaml
- service.yaml
- pod.yaml
```
will be
```
# some descriptions for the patches
patches:
- service.yaml
- pod.yaml
```

Reply:
ok. Thanks.